src/misc/fs_utils_posix.cpp

```diff
@@ -354,4 +357,104 @@ uint16_t local_drive_set_attributes(const std_fs::path& path,
 	return status ? DOSERR_NONE : DOSERR_ACCESS_DENIED;
 }
 
+// POSIX does not guarantee to read or write all bytes requested at once.
+// On Linux, it will read the entire chunk in one call (assuming it's a normal file).
+// On Windows, it does not always do so (although we're not using these functions on Windows).
+// On Mac and the various BSDs, who knows. So be safe and do a loop.
+NativeFileReturn local_drive_read_file(const NativeFileHandle handle, uint8_t *buffer, const uint32_t bytes_requested)
+{
+	NativeFileReturn ret = {};
+	ret.bytes = 0;
+	ret.error = false;
+	while (ret.bytes < bytes_requested) {
+		const auto bytes = read(handle, buffer + ret.bytes, bytes_requested - ret.bytes);
+		if (bytes <= 0) {
+			ret.error = bytes < 0;
+			break;
+		}
+		ret.bytes += bytes;
+	}
+	return ret;
+}
+
+NativeFileReturn local_drive_write_file(const NativeFileHandle handle, const uint8_t *buffer, const uint32_t bytes_requested)
+{
+	NativeFileReturn ret = {};
+	ret.bytes = 0;
+	ret.error = false;
+	while (ret.bytes < bytes_requested) {
+		const auto bytes = write(handle, buffer + ret.bytes, bytes_requested - ret.bytes);
+		if (bytes <= 0) {
+			ret.error = bytes < 0;
+			break;
+		}
+		ret.bytes += bytes;
+	}
+	return ret;
+}
+
+int32_t local_drive_seek_file(const NativeFileHandle handle, const int32_t offset, const NativeFileSeek type)
+{
+	int posix_seek_type = SEEK_SET;
+	switch(type) {
```

Comment:
Space after `switch`.

Reply:
Well, just reformat the whole file; formatting is off.

Reply:
formatted